synapse/handlers/ui_auth/checkers.py

```diff
@@ -39,6 +39,7 @@ def is_enabled(self) -> bool:
         Returns:
             True if this login type is enabled.
         """
+        raise NotImplementedError()
```

Comment:
Should this be `abstractmethod`?

Reply:
Probably. The docstring even says this is an `Abstract base class`.

Reply:
I had to add two new linter ignores here, to make UserInteractiveAuthChecker an ABC, but having done so mypy did detect an incomplete subclass in the tests. The commit is 8395cc89f9 --- what do you think of the change?

Reply:
I think that's a change for the positive?